platform/darwin/src/MGLStyleValue.h

```diff
@@ -62,32 +63,70 @@ MGL_EXPORT
+ @return An `MGLCameraStyleFunction` object with the given interpolation mode,
+    camera stops, and options.
+ */
++ (instancetype)valueWithInterpolationMode:(MGLInterpolationMode)interpolationMode cameraStops:(NS_DICTIONARY_OF(id, MGLStyleValue<T> *) *)cameraStops options:(nullable NS_DICTIONARY_OF(MGLStyleFunctionOption, id) *)options;
 
-// TODO: API docs
-+ (instancetype)valueWithInterpolationMode:(MGLInterpolationMode)interpolationMode sourceStops:(nullable NS_DICTIONARY_OF(id, MGLStyleValue<T> *) *)stops attributeName:(NSString *)attributeName options:(nullable NS_DICTIONARY_OF(MGLStyleFunctionOption, id) *)options;
+/**
+ Creates and returns an `MGLSourceStyleFunction` object representing a source function.
+
+ @param interpolationMode The mode used to interpolate property values over a 
+    range of feature attribute values.
+ @param sourceStops A dictionary associating feature attributes with style values.
+ @param attributeName Specifies the feature attribute to take as the function 
+    input.
+ @param options A dictionary containing `MGLStyleFunctionOption` values that
+    specify how a function is applied.
+ @return An `MGLSourceStyleFunction` object with the given interpolation mode,
+    source stops, attribute name, and options.
+ */
++ (instancetype)valueWithInterpolationMode:(MGLInterpolationMode)interpolationMode sourceStops:(nullable NS_DICTIONARY_OF(id, MGLStyleValue<T> *) *)sourceStops attributeName:(NSString *)attributeName options:(nullable NS_DICTIONARY_OF(MGLStyleFunctionOption, id) *)options;
 
-// TODO: API docs
-+ (instancetype)valueWithInterpolationMode:(MGLInterpolationMode)interpolationMode compositeStops:(NS_DICTIONARY_OF(id, NS_DICTIONARY_OF(id, MGLStyleValue<T> *) *) *)stops attributeName:(NSString *)attributeName options:(nullable NS_DICTIONARY_OF(MGLStyleFunctionOption, id) *)options;
+/**
+ Creates and returns an `MGLCompositeStyleFunction` object representing a composite
+ function made up of one or more outer stops interpreted as a camera function 
+ with a single inner stop at each zoom level that is interpreted as a source 
+ function.
+
+ @param interpolationMode The mode used to interpolate property values over a 
+    range of feature attribute values for each outer zoom level.
+ @param sourceStops A dictionary associating feature attributes with style values.
+ @param attributeName Specifies the feature attribute to take as the function
+    input.
+ @param options A dictionary containing `MGLStyleFunctionOption` values that
+    specify how a function is applied.
+ @return An `MGLCompositeStyleFunction` object with the given interpolation mode,
+    composite stops, attribute name, and options.
+ */
++ (instancetype)valueWithInterpolationMode:(MGLInterpolationMode)interpolationMode compositeStops:(NS_DICTIONARY_OF(id, NS_DICTIONARY_OF(id, MGLStyleValue<T> *) *) *)compositeStops attributeName:(NSString *)attributeName options:(nullable NS_DICTIONARY_OF(MGLStyleFunctionOption, id) *)options;
```

Comment:
This was me doing a poor job of trying to clarify the opposite: "composite" in this case means a composite of stops dictionaries not functions (see discussion in https://github.com/mapbox/mapbox-gl-native/pull/7596#issuecomment-276530479). In https://github.com/mapbox/mapbox-gl-native/pull/7943/commits/d46559a5ff8791af9fb0b39cce20d31c13a57bb8, I punted on trying to explain this in the class factory method in favor of the explanation in the `MGLCompositeStyleFunction` `stops` property